File: src/2015/day1/day1_part2.c
#include <stdio.h>
int main() {
  FILE *file = fopen("2015_day1.txt", "r");
  int floor = 0;
  int ch;
  int pos = 0;
  while ((ch = getc(file)) != EOF) {
    pos++;
    if (ch == '(') {
      floor++;
    } else if (ch == ')') {
      floor--;
    }
    if (floor == -1){
      printf("Basement reached at char position %d\n", pos);
      break;
    }
  }

  fclose(file);
  return 0;
}

File: src/2015/day1/day1_part2.go
package main

import(
	"fmt"
	"os"
)
func main(){
	content, _ := os.ReadFile("./2015_day1.txt")
	floor :=0 
	for i, r := range content{
		if(r == '('){
		 	floor++
		}
		if(r == ')'){
		 	floor--
		}
		if(floor == -1){
			fmt.Println(i + 1)
			break
		}
	}
}